x-pack/legacy/plugins/apm/public/components/shared/LocalFilters/Filter.tsx

```diff
@@ -0,0 +1,209 @@
+
+const Popover = styled(EuiPopover).attrs({
+  anchorClassName: 'anchor'
+})`
+  .anchor {
+    display: block;
+  }
+`;
+
+const SelectContainer = styled.div`
+   {
+    width: 300px;
+  }
+`;
+
+const Counter = styled.div`
+  border-radius: ${theme.euiBorderRadius};
+  background: ${theme.euiColorLightShade};
+  padding: 0 ${theme.paddingSizes.xs};
+`;
+
+const BadgeText = styled.div`
+   {
+    display: inline-block;
+    max-width: 140px;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    vertical-align: middle;
+  }
+`;
+
+interface Props {
+  name: string;
+  title: string;
+  options: Array<{
+    name: string;
+    count: number;
+  }>;
+  onChange: (value: string[]) => void;
+  value: string[];
+}
+
+const Filter: React.FC<Props> = props => {
+  const { name, title, options, onChange, value } = props;
+
+  const [showPopover, setShowPopover] = useState(false);
+
+  const toggleShowPopover = useCallback(() => setShowPopover(!showPopover), [
+    showPopover
+  ]);
+
+  const button = (
+    <EuiButtonEmpty
+      flush="left"
+      color="text"
+      onClick={() => setShowPopover(!showPopover)}
+      style={{ display: 'block' }}
+    >
+      <EuiTitle size="xxxs" textTransform="uppercase">
+        <h4>{title}</h4>
+      </EuiTitle>
+    </EuiButtonEmpty>
+  );
+
+  // This will break when options or value changes,
+  // but
+  const items = useMemo(
+    () =>
+      options.map(
+        option =>
+          ({
+            label: option.name,
+            append: (
+              <Counter>
+                <EuiText size="xs">{option.count}</EuiText>
+              </Counter>
+            ),
+            ...(value.includes(option.name)
+              ? { checked: 'on' as 'on' | 'off' | 'undefined' }
+              : {})
+          } as {
+            label: string;
+            append?: React.ReactNode;
+            checked?: 'on' | 'off' | undefined;
+          })
+      ),
+    [value, options]
+  );
+
+  const [visibleOptions, setVisibleOptions] = useState(items);
+
+  const onApply = useCallback(() => {
+    const newValue = visibleOptions
+      .filter(option => option.checked === 'on')
+      .map(option => option.label);
+
+    setShowPopover(false);
+    onChange(newValue);
+  }, [onChange, visibleOptions]);
+
+  useEffect(() => {
+    setVisibleOptions(items);
+  }, [items]);
+
+  const badges = value.length ? (
+    <>
+      <EuiFlexGrid gutterSize="s">
+        {value.map(val => (
+          <EuiFlexItem key={val} grow={false}>
+            <button
+              type="button"
+              onClick={() => {
+                onChange(value.filter(v => val !== v));
+              }}
+            >
+              <EuiBadge color="hollow">
+                <BadgeText>{val}</BadgeText>
+                <EuiIcon type="cross"></EuiIcon>
+              </EuiBadge>
+            </button>
+          </EuiFlexItem>
+        ))}
+      </EuiFlexGrid>
+      <EuiSpacer size="s"></EuiSpacer>
+    </>
+  ) : (
+    []
```

Comment:
Shouldn't this be `null` instead of `[]`?

Reply:
Yep!